File: Stage19/C/Problem2447.c
#include <stdio.h>

char board[2200][2200];

void draw(int x, int y, int size) {
    if (size == 1) return;

    int newSize = size / 3;

    for (int i = x + newSize; i < x + 2 * newSize; i++) {
        for (int j = y + newSize; j < y + 2 * newSize; j++) {
            board[i][j] = ' ';
        }
    }

    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            if (i == 1 && j == 1) continue;
            draw(x + i * newSize, y + j * newSize, newSize);
        }
    }
}

int main(void) {
    int N;
    scanf("%d", &N);

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            board[i][j] = '*';
        }
    }

    draw(0, 0, N);

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            putchar(board[i][j]);
        }
        putchar('\n');
    }

    return 0;
}

File: Stage19/Python/Problem2447.py
import sys
input = sys.stdin.readline

def draw(board, x, y, size):
    if size == 1:
        return

    newSize = size // 3

    for i in range(x + newSize, x + 2 * newSize):
        for j in range(y + newSize, y + 2 * newSize):
            board[i][j] = ' '

    for i in range(3):
        for j in range(3):
            if i == 1 and j == 1: 
                continue
            draw(board, x + i * newSize, y + j * newSize, newSize)

def main():
    N = int(input())
    board = [['*'] * N for _ in range(N)]

    draw(board, 0, 0, N)

    print("\n".join("".join(row) for row in board))

if __name__ == "__main__":
    main()